### PR TITLE
Upgrade to Scala 3.0.0-M2

### LINKED ISSUE #36: Publishing for 3.0.0-M2
Not a request, just creating this for tracking to make sure @eed3si9n doesn't already have it working :)

I've started a PR #35 but I'm flying blind with Scala 3 macros, so expect a lot of stupid stuff to happen in that PR.

### PR DESCRIPTION
**Pull request by keynmol**
Closes #36 

~Just hoping to get CI confirm that my local failures are my fault.~

~I don't know what I'm doing with macros, so this will take a while :)~

This all seems to be okay, I think. I've tested it locally with a branch of Weaver, and expecty passed all tests

### CODE REVIEW

**Comment by eed3si9n** [comment]
See also https://github.com/scala/nanotest-strawman

**Comment by keynmol** [comment]
Thanks @eed3si9n, I don't know if you meant it as "look at this for inspiration", but I took it as such and managed to fix the rest of the macro :)

And I'm hoping to get 3.0.0-M2 support out because we use expecty in Weaver Test, and wanna reach for that uber-versions-matrix

File: project/plugins.sbt [hunk 1]
```diff
@@ -1,10 +1,10 @@
 val scalaJSVersion =
-  Option(System.getenv("SCALAJS_VERSION")).getOrElse("1.2.0")
+  Option(System.getenv("SCALAJS_VERSION")).getOrElse("1.3.1")
```
Comment (keynmol) [inline project/plugins.sbt:2]: Forced by 3.0.0-M2

**Comment by keynmol** [comment]
Hey @eed3si9n is there anything else you'd want me to change here to get this merged and released?

And on a meta note - I see the version of this macro getting more features in the scala-verify. Does this mean expecty will at some point stop being maintained/released?

**Review by eed3si9n — APPROVED** [review]

File: project/plugins.sbt (hunk 1 above)
Comment (keynmol) [reply project/plugins.sbt:2]: I *think* this means next version needs to be 0.15.1, but I always get confused about Scala.js bincompat guarantees

**Comment by eed3si9n** [comment]
> And on a meta note - I see the version of this macro getting more features in the scala-verify. Does this mean expecty will at some point stop being maintained/released?

I think it depends on how popular Verify gets. It would certainly be more convenient if Verify and Expecty could converge so we don't have to maintain both. The good thing about Verify (at least for now) is that it's part of the Dotty Community Build so the macro patches are created by Dotty team - https://github.com/scala/nanotest-strawman/pull/117

**Pull request merged**